core/src/banking_stage/consumer.rs

```diff
@@ -392,14 +392,54 @@ impl Consumer {
         bank: &Arc<Bank>,
         txs: &[SanitizedTransaction],
         chunk_offset: usize,
+    ) -> ProcessTransactionBatchOutput {
+        // No filtering before QoS - transactions should have been sanitized immediately prior to this call
+        let pre_results = std::iter::repeat(Ok(()));
+        self.process_and_record_transactions_with_pre_results(bank, txs, chunk_offset, pre_results)
+    }
+
+    pub fn process_and_record_aged_transactions(
+        &self,
+        bank: &Arc<Bank>,
+        txs: &[SanitizedTransaction],
+        max_slot_ages: &[Slot],
+    ) -> ProcessTransactionBatchOutput {
+        // Need to filter out transactions since they were sanitized earlier.
+        // This means that the transaction may cross and epoch boundary (not allowed),
+        //  or account lookup tables may have been closed.
+        let pre_results = txs
+            .iter()
+            .zip(max_slot_ages)
+            .map(|(tx, max_slot_age)| {
+                if *max_slot_age < bank.slot() {
+                    // Attempt re-sanitization after epoch-cross.
+                    // Only need to verify pre-compiles.
+                    tx.verify_precompiles(&bank.feature_set)?;
+                }
+                let lookup_tables = tx.message().message_address_table_lookups();
+                if !lookup_tables.is_empty() {
+                    bank.load_addresses(lookup_tables)?;
+                }
+                Ok(())
+            })
+            .collect_vec();
+        self.process_and_record_transactions_with_pre_results(bank, txs, 0, pre_results.into_iter())
```

Comment:
nit: i think we can remove this redundant back-and-force of iter => vec => iter by just simply passing `impl Iterator<Item ...>`...:

```rust
let pre_results = ...
    .collect_vec();
self.process_and_record_transactions_with_pre_results(..., pre_results.into_iter());
```

Reply:
done: [9871481](https://github.com/solana-labs/solana/pull/30970/commits/98714817efbf60bc7fe9f4f752af38aaf6dd5dd7)

I think when I had mutability within the pre_results iteration, I could not do it. With mutability gone, I can.